pkg/kubelet/dockertools/manager.go

```diff
@@ -280,8 +280,15 @@ func (dm *DockerManager) inspectContainer(dockerID, containerName, tPath string)
 		result.status.State.Running = &api.ContainerStateRunning{
 			StartedAt: util.NewTime(inspectResult.State.StartedAt),
 		}
-		if containerName == PodInfraContainerName && inspectResult.NetworkSettings != nil {
-			result.ip = inspectResult.NetworkSettings.IPAddress
+		if dm.networkPlugin.Name() != network.DefaultPluginName && containerName == PodInfraContainerName {
+			netStatus, err := dm.networkPlugin.Status(pod.Namespace, pod.Name, kubeletTypes.DockerID(dockerID))
+			if err == nil {
+				result.ip = netStatus.IPv4.String()
```

Comment:
@rajatchopra, @mrunalp - yes, works for me

one minor follow-up: how do we identify if IPv4 or IPv6 is valid? `result.ip` could be filled in with a valid address (for the case I was alluding for ipv6 deployment).